File: TextosJapones/src/main/java/org/jisho/textosJapones/components/TableViewNoSelectionModel.java
package org.jisho.textosJapones.components;

import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.scene.control.TableColumn;
import javafx.scene.control.TablePosition;
import javafx.scene.control.TableView;

public class TableViewNoSelectionModel<T> extends TableView.TableViewSelectionModel<T> {

	public TableViewNoSelectionModel(TableView<T> tableView) {
		super(tableView);
	}

	@Override
	public ObservableList<TablePosition> getSelectedCells() {
		return FXCollections.emptyObservableList();
	}

	@Override
	public void selectLeftCell() {

	}

	@Override
	public void selectRightCell() {

	}

	@Override
	public void selectAboveCell() {

	}

	@Override
	public void selectBelowCell() {

	}

	@Override
	public void clearSelection(int i, TableColumn tableColumn) {

	}

	@Override
	public void clearAndSelect(int i, TableColumn tableColumn) {

	}

	@Override
	public void select(int i, TableColumn tableColumn) {

	}

	@Override
	public boolean isSelected(int i, TableColumn tableColumn) {
		return false;
	}

	@Override
	public ObservableList<Integer> getSelectedIndices() {
		return FXCollections.emptyObservableList();
	}

	@Override
	public ObservableList getSelectedItems() {
		return FXCollections.emptyObservableList();
	}

	@Override
	public void selectIndices(int i, int... ints) {

	}

	@Override
	public void selectAll() {

	}

	@Override
	public void clearAndSelect(int i) {

	}

	@Override
	public void select(int i) {

	}

	@Override
	public void select(Object o) {

	}

	@Override
	public void clearSelection(int i) {

	}

	@Override
	public void clearSelection() {

	}

	@Override
	public boolean isSelected(int i) {
		return false;
	}

	@Override
	public boolean isEmpty() {
		return false;
	}

	@Override
	public void selectPrevious() {

	}

	@Override
	public void selectNext() {

	}

	@Override
	public void selectFirst() {

	}

	@Override
	public void selectLast() {

	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/components/TableViewNoSelectionModel.kt
package br.com.fenix.processatexto.components

import javafx.collections.FXCollections
import javafx.collections.ObservableList
import javafx.scene.control.TableColumn
import javafx.scene.control.TablePosition
import javafx.scene.control.TableView


class TableViewNoSelectionModel<T>(tableView: TableView<T>) : TableView.TableViewSelectionModel<T>(tableView) {

    override fun getSelectedCells(): ObservableList<TablePosition<Any, Any>> = FXCollections.emptyObservableList()

    @Override
    override fun selectLeftCell() {
    }

    @Override
    override fun selectRightCell() {
    }

    @Override
    override fun selectAboveCell() {
    }

    @Override
    override fun selectBelowCell() {
    }

    @Override
    override fun clearSelection(i: Int, tableColumn: TableColumn<T, *>) {
    }

    @Override
    override fun clearAndSelect(i: Int, tableColumn: TableColumn<T, *>) {
    }

    @Override
    override fun select(i: Int, tableColumn: TableColumn<T, *>) {
    }

    @Override
    override fun isSelected(i: Int, tableColumn: TableColumn<T, *>): Boolean {
        return false
    }

    override fun getSelectedIndices() = FXCollections.emptyObservableList<Int>()

    override fun getSelectedItems() = FXCollections.emptyObservableList<T>()

    @Override
    override fun selectIndices(i: Int, vararg ints: Int) {
    }

    @Override
    override fun selectAll() {
    }

    @Override
    override fun clearAndSelect(i: Int) {
    }

    @Override
    override fun select(i: Int) {
    }

    @Override
    override fun select(obj: T) {
    }

    @Override
    override fun clearSelection(i: Int) {
    }

    @Override
    override fun clearSelection() {
    }

    @Override
    override fun isSelected(i: Int): Boolean {
        return false
    }

    override fun isEmpty() = false

    @Override
    override fun selectPrevious() {
    }

    @Override
    override fun selectNext() {
    }

    @Override
    override fun selectFirst() {
    }

    @Override
    override fun selectLast() {
    }
}